lib/src/main/java/uk/gov/di/ipv/cri/passport/library/helpers/KmsSigner.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.di.ipv.cri.passport.library.helpers;
+
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+import com.amazonaws.services.kms.model.MessageType;
+import com.amazonaws.services.kms.model.SignRequest;
+import com.amazonaws.services.kms.model.SignResult;
+import com.amazonaws.services.kms.model.SigningAlgorithmSpec;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.jca.JCAContext;
+import com.nimbusds.jose.util.Base64URL;
+import uk.gov.di.ipv.cri.passport.library.annotations.ExcludeFromGeneratedCoverageReport;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Set;
+
+import static com.nimbusds.jose.JWSAlgorithm.RS256;
+
+public class KmsSigner implements JWSSigner {
+
+    private final AWSKMS kmsClient;
+
+    private static final Base64.Encoder b64UrlEncoder = Base64.getUrlEncoder();
+    private final JCAContext jcaContext = new JCAContext();
+    private final String keyId;
+
+    @ExcludeFromGeneratedCoverageReport
+    public KmsSigner(String keyId) {
+        this.keyId = keyId;
+        this.kmsClient = AWSKMSClientBuilder.defaultClient();
+    }
+
+    public KmsSigner(String keyId, AWSKMS kmsClient) {
+        this.keyId = keyId;
+        this.kmsClient = kmsClient;
+    }
+
+    @Override
+    public Base64URL sign(JWSHeader header, byte[] signingInput) throws JOSEException {
+        byte[] signingInputHash;
+
+        try {
+            signingInputHash = MessageDigest.getInstance("SHA-256").digest(signingInput);
+        } catch (NoSuchAlgorithmException e) {
+            throw new JOSEException(e.getMessage());
+        }
+
+        SignRequest signRequest =
+                new SignRequest()
+                        .withSigningAlgorithm(
+                                SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256.toString())
```

Comment:
We're moving more towards using EC keys rather than RSA, to be more inline with what SPOT and (I believe) the KBV team are doing. This will need updating to `SigningAlgorithmSpec.ECDSA_SHA_256`.

Reply:
Nice! Updated.